how-to-build.md

```diff
@@ -2,46 +2,58 @@ This file explains how to build and install the Realm core library.
 
 ## Prerequisites
 
-To build the Realm core library, you need the standard set of build
-tools. This includes a C/C++ compiler and GNU make. Realm is
-thoroughly tested with both GCC and Clang. It is known to work with
-GCC 4.2 and newer, as well as with Clang 3.0 and newer.
+To build the Realm core library, you need CMake 3.4 or newer and a
+standard set of build tools. This includes a C/C++ compiler and a
+build system like GNU make. Realm is thoroughly tested with both GCC
+and Clang. It is known to work with GCC 4.9 and newer, as well as
+with Clang 8.0 and newer. Your compiler must support C++14.
```

Comment:
"8.0" is part of a versioning scheme used by the version of Clang that Apple ships. The latest version using LLVM's versioning schema is 4.0. I don't know why Apple invented their own version numbering like this. I'm not sure what the best way to concisely clarify this is either.